### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by samdoran**
## Description

Update cryptography to address some CVEs.
Remove unused packages:
 - `importlib-metadata`
 - `pint`
 - `ujson`

Separate direct and indirect decencies in `Pipfile` for clarity.

### CODE REVIEW

**Comment by samdoran** [comment]
This still results in an image that snyk claims has an outdated `pip` with some [medium vulnerabilities](https://quay.io/repository/samdoran/koku/manifest/sha256:d5c76a44f959c9f429307bf88f5048a997ff3770bf782031cd75ae0d123a48dc?tab=vulnerabilities) because the venv is created in a separate layer than `pip` is upgraded. The `pip` version in the final layer is the latest.

I modified the Dockerfile slightly to create the venv, then updated `pip` and `setuptools` in the same layer. That [gets us green](https://quay.io/repository/samdoran/koku/manifest/sha256:fe3edbbed58853505de67ed28b65e7ecb2662dd4cebfd0dc144d678cbb24db97?tab=vulnerabilities).

<details>
  <summary>Diff</summary>
  
```patch
diff --git Dockerfile Dockerfile
index 8b0a39fa..a308c94b 100644
--- Dockerfile
+++ Dockerfile
@@ -70,12 +70,11 @@ ARG USER_ID=1000
 # Create a Python virtual environment for use by any application to avoid
 # potential conflicts with Python packages preinstalled in the main Python
 # installation.
-RUN python3.9 -m venv /pipenv-venv
+RUN python3.9 -m venv /pipenv-venv \
+    && /pipenv-venv/bin/python -m pip install --upgrade pip setuptools \
+    && /pipenv-venv/bin/python -m pip install pipenv
+
 ENV PATH="/pipenv-venv/bin:$PATH"
-# Install pipenv into the virtual env
-RUN \
-    pip install --upgrade pip && \
-    pip install pipenv
 
 WORKDIR ${APP_ROOT}
 

```
</details>

**Comment by maskarb** [comment]
I'm not sure how you updated this Pipfile.lock, but it is not correct. Anytime a dependency in the lock file changes, the hash at the top of the file also changes. 

Also, an oversight: the Pipfile.lock needs to be copied in the dockerfile. The image did not build correctly during smokes because it relied on caching.

The reason cryptography is out of date is because it is a dependency of OCI, and OCI has it pinned to a lower version. Maybe things have changed since the last time I have looked at this.

**Comment by samdoran** [comment]
I think it did build successfully, though.

```
23:46:53 #16 naming to quay.io/cloudservices/koku:pr-4248-0e0fe4b done
23:46:53 #16 naming to quay.io/cloudservices/koku:pr-4248-latest done
23:46:53 #16 DONE 0.1s
23:46:53 ++ set +x
23:46:53 ++ docker push quay.io/cloudservices/koku:pr-4248-0e0fe4b
23:46:53 The push refers to repository [quay.io/cloudservices/koku]
```

The image on Quay looks correct. Not sure why the smoke tests failed.

**Comment by samdoran** [comment]
Ended up needing to update to a newer version of `oci` that requires a new enough version of `cryptography` to get our image clean.

**Comment by samdoran** [comment]
I'm not sure what our cadence and process is for updating all our dependencies. I haven't worked extensively in a project that has this many external dependencies, so I'm still trying to work out how best to go about keeping this up to date, keeping the application stable, and getting us bogged down putting out fires due to library updates (_stares at Azure SDK_).

**Comment by maskarb** [comment]
None of us really have any time to go review all release notes for all of these packages. We have unit tests and smoke tests to catch breaking issues. Sometimes one will get through, but we can always react to those if it happens.

Let's just relock the file, run our test suite, and see what happens.

If we tried to review release notes every time a library updated, we would spend half our time reading notes instead of pushing out new code.

We can rely on dependabot to let us know when a Python package has a vulnerability and try to address them when they arise. Normally dependabot is good enough to handle creating the dependency update PRs for us. Cryptography and OCI were different because OCI was preventing us from updating cryptography and dependabot wasn't able to open the PR for us.

**Comment by samdoran** [comment]
I'll update all the thing then.

File: Pipfile [hunk 1]
```diff
@@ -4,84 +4,86 @@ verify_ssl = true
...
+django-cors-headers = ">=3.1"
+django-environ = ">=0.4"
+django-extensions = ">=2.2"
+django-filter = ">=2.2"
+django-prometheus = ">=1.1"
+django-redis = ">=4.10"
+django-tenant-schemas = ">=1.10"
+djangorestframework = ">=3.11,!=3.14.0"
+djangorestframework-csv = ">=2.1"
+flake8 = ">=3.7"
```
Comment (maskarb) [inline Pipfile:32]: didn't we remove this?

Comment (samdoran) [reply Pipfile:32]: Yup. Bad merge.

**Review by maskarb — APPROVED** [review]

**Comment by samdoran** [comment]
Successful smoke test: https://ci.ext.devshift.net/job/project-koku-koku-pr-check/8886/

**Pull request merged**